vignettes/complete-multiverse-analysis.Rmd

```diff
@@ -232,32 +233,137 @@ Since the multiverse object has already been created and the one parameter has a
 
-```{r, eval = FALSE}
 universe_parameter_assignment <- list(
     menstrual_calculation = "mc_option",
     relationship_status = "rs_option3",
     cycle_length = "cl_option2",
     certainty = "cer_option1",
     fertile = "fer_option4"
   )
+
 M %>% execute_default( universe_parameter_assignment ) %>%
   select(ComputedCycleLength, NextMenstrualOnset, CycleDay, Relationship, Fertility, everything()) %>%
   head()
 ```
+
+
+## Specifying conditions in the multiverse analysis
+In a multiverse analysis, it may occur that the value of one variable might depend on the value of another variable defined previously. 
+For example, in our example, we are excluding participants based on their *cycle length*. This can be done in two ways: we can use the values of the variable,`ComputedCycleLength` or `ReportedCycleLength`. If we are using `ComputedCycleLength` to exclude participants, this means that we should not calculate the variable `NextMenstrualOnset` (date for the onset of the next menstrual cycle) using the `ReportedCycleLength` value. Similarly, if we are using `ReportedCycleLength` to exclude participants it is inconsistent to calculate `NextMenstrualOnset` using `ComputedCycleLength`.
+
+We should be able to express these conditions in the multiverse. We can do this in two ways:
+1. `%when%`: when declaring a branch, we can use this operator to specify the condition as `A %when% B`
+2. `branch_assert`: the conditional $A | B$ is also referred to as the connective $A \implies B$. This has the meaning "if A is true, then B is also true" and is an abbreviation for $\neg A | B$
+
+#### The %when% operator
+There are two ways in which you can specify the `%when%` operator. The first is to specify it at the end of the branch. This will work even if you omit the branch option name.
+
+```{r, eval = FALSE}
+M$df <- data.raw.study2  %>%
+    mutate( ComputedCycleLength = StartDateofLastPeriod - StartDateofPeriodBeforeLast ) %>%
+    mutate(NextMenstrualOnset = branch(menstrual_calculation,
+            "mc_option1" ~ (StartDateofLastPeriod + ComputedCycleLength) %when% (cycle_length != "cl_option3"),
+            "mc_option2" ~ (StartDateofLastPeriod + ReportedCycleLength) %when% (cycle_length != "cl_option2"),
+            "mc_option3" ~ StartDateNext)
+    )
+```
+
+The other is to specify it at the head of the branch, right after the option name:
+
+```{r, eval = FALSE}
+M$df <- df  %>%
+    mutate(NextMenstrualOnset = branch(menstrual_calculation,
+            "mc_option1" %when% (cycle_length != "cl_option3") ~ StartDateofLastPeriod + ComputedCycleLength,
+            "mc_option2" %when% (cycle_length != "cl_option2") ~ (StartDateofLastPeriod + ReportedCycleLength),
+            "mc_option3" ~ StartDateNext)
+    )
+```
+
+#### The `branch_assert()` function
+Within the branch assert function, the user needs to specify the classical logic specification $\neg A | B$:
```

Comment:
Why? Can't they provide arbitrary conditionals?

Reply:
I guess. I don't see why not --- as long as it is a valid conditional. We are not doing anything except extracting their expression and running it as a filter on the multiverse table.

Reply:
Yeah, then I wouldn't word it as "the user needs to". Maybe first say they can specify any filtering condition (this motivates why it exists: it is more general than the %when% syntax) and then show how to translate the previous example